### ISSUE TITLE
Execution of 'new' PIConGPU with openPMD fails with `unknown datatype (switchType)`

### ISSUE THREAD

**Issue by Anton-Le** [ComputationalRadiationPhysics/picongpu#3732]
I have attempted to run a simulation with newest (baf0da494bd4d7e432033dd61242f4efcba1d39d) PIC and openPMD 0.14.1, 0.14.2/dev (040a9b0) on JWB and hemera - to no avail.

The simulations will consistently fail on the first attempt to write data with openPMD. The errors are identical, whether I try to write particle & field data or a checkpoint.

Excerpts from `stderr` for the above commit of PIC:
`Unhandled exception of type 'St13runtime_error' with message 'Internal error: Encountered unknown datatype (switchType) ->35', terminating`
```
[cupla] Error: </p/project/pwfaradiation/lebedev1/PIConGPU_SourcesAndLibs_Aug21/picongpu-dev/include/pmacc/../pmacc/memory/buffers/HostBufferIntern.hpp>:71 
[cupla] Error: </p/project/pwfaradiation/lebedev1/PIConGPU_SourcesAndLibs_Aug21/picongpu-dev/include/pmacc/../pmacc/memory/buffers/Buffer.hpp>:69
```

```
[cupla] Error: </p/project/pwfaradiation/lebedev1/PIConGPU_SourcesAndLibs_Aug21/picongpu-dev/include/pmacc/../pmacc/me
mory/buffers/Buffer.hpp>:69 
 6 0x0000000000e9f768 _ZThn56_N8picongpu9ParticlesIN5pmacc4meta6StringIJLc119ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0E
Lc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0EEEEN5boost3mpl6v_itemINS_11chargeRatioINS_20ChargeRatioElectronsENS1_13pmacc_isAliasEEENS7_INS_9massRatioINS_18MassRatioElectronsESA_EENS7_INS_12densityRatioINS_17DensityRatioBunchESA_EENS7_INS_7currentINS_13currentSolver3EmZINS_9particles6shapes3TSCENSJ_8strategy16CachedSupercellsEEESA_EENS7_INS_13interpolationINS_28FieldToParticleInterpolationISN_NS_30AssignedTrilinearInterpolationEEESA_EENS7_INS_5shapeISN_SA_EENS7_INS_14particlePusherINSL_6pusher9CompositeINS10_12AccelerationENS10_5BorisENS10_38CompositeBinarySwitchActivationFunctorILj22838EEEEESA_EENS6_7vector0IN4mpl_2naEEELi0EEELi0EEELi0EEELi0EEELi0EEELi0EEELi0EEENS7_INS_13radiationMaskENS7_INS_13momentumPrev1ENS7_INS_9weightingENS7_INS_8momentumENS7_INS_8positionINS_12position_picESA_EES1B_Li0EEELi0EEELi0EEELi0EEELi0EEEED0Ev()  ???:0
```

My guess _was_ that the problem lies with the `CompositeBinarySwitchActivationFunctor`, i.e., the functor that allows me to 
switch from acceleration to actual propagation w/o checkpointing first. However, removing said functor from `species.param` and `particle.param` did **not** solve the issue.

These - or largely similar - errors have also been observed on hemera.

Library configurations and config/cmake outputs are attached. 
[LibConfig.md](https://github.com/ComputationalRadiationPhysics/picongpu/files/7026201/LibConfig.md)

**Comment by sbastrakov** [comment]
I can confirm that there are issues observed on our dev system as well.

**Comment by Anton-Le** [comment]
Temporarily I can circumvent this issue in my simulations, by reverting to a PIConGPU version which permits ADIOS1 usage, e.g. 3bab80f97b609c7c29931952d11cef6c8d5765ad

**Comment by sbastrakov** [comment]
@Anton-Le sorry it took a bit long. I've now tried your setup on Hemera fwkt_v100 partition with its `openpmd/0.13.2-cuda112-adios271` module (standard in our profile in `dev` for that partition), and it seems to work.

**Comment by Anton-Le** [comment]
Weird. I will re-try it.

**Comment by franzpoeschel** [comment]
I see, you found the issue during my holiday
Using chars for this purpose is probably for the best, yeah

**Comment by PrometheusPi** [comment]
@sbastrakov Did @sbastrakov's pull request solve the issue?

**Comment by sbastrakov** [comment]
I guess you meant to tag @Anton-Le . As far as i can see, there was still some issue observed on JUWELS, not clear if related or not and hard to check now after the allocation ended.

**Comment by Anton-Le** [comment]
Unfortunately I could not test this on JWB anymore. I will launch the same test on hemera today to check the problem. The problem on JWB was that, although I could _write_ a checkpoint with the fix, a restart from said checkpoint failed.

**Comment by sbastrakov** [comment]
@Anton-Le in case you are using a moving windonw, it could have been same issue as in #3899 . The fix #3902 is soon to be merged. The bug could have caused not just weird results, but also crashes.

**Comment by sbastrakov** [comment]
Btw that would have also explained why my small reproducer runs did not show the error, it occurs only after the window have actually slided (i.e. passed at least one local domain size)

**Comment by Anton-Le** [comment]
That could very well be it! I'm normally using a moving window.

**Comment by PrometheusPi** [comment]
Did you output pngs @Anton-Le? If yes, you can check if the bug was triggered after restart.

**Comment by sbastrakov** [comment]
Iirc it was crashing during the restart?
Btw now the fix is merged to `dev`.

**Comment by psychocoderHPC** [comment]
I will close this issue, it should be fixed with #3890 and we do not have any compute time on JUWELS to reproduce the issue.
